src/client/content_cao.cpp

```diff
@@ -745,9 +745,6 @@ void GenericCAO::addToScene(ITextureSource *tsrc, scene::ISceneManager *smgr)
 		m_meshnode = m_smgr->addMeshSceneNode(mesh, m_matrixnode);
 		m_meshnode->grab();
 		mesh->drop();
-		// Set it to use the materials of the meshbuffers directly.
-		// This is needed for changing the texture in the future
-		m_meshnode->setReadOnlyMaterials(true);
```

Comment:
Why is this no longer needed/desired?
Is it because we're setting the texture later?

Reply:
This is not desired, because the scene node blocks overrides of materials via `getMaterial` when rendering, and shadow renderer cannot attach the right shader to transform and render the shadow. This was the cause of phantom shadows I described earlier.

Changing textures works just fine with node-level materials, as could be seen in the code for "cube" entity type. I adjusted the code in `setTextures` to work with node-level materials and not mesh buffer materials.